File: game/tv_show/instructions.py
from config import Config
from phone_events import PhoneEvents
from state import State
from tv_show.attract import Attract
from tv_show.playing import Playing
from tv_show.tv_show_resources import TvShowResources


class Instructions(State):
    def process_events(self, phone_events: PhoneEvents):
        if phone_events.hungup:
            return Attract

        if self.get_state_time() > Config.INSTRUCTIONS_TIMEOUT:
            return Playing

        return None


    def render(self, screen):
        screen.blit(TvShowResources.instructions["Forest"], (0, 0))

File: game-web/src/tvShow/instructions.ts
/** Port of game/tv_show/instructions.py */

import { State, type StateClass } from '../state';
import { Config } from '../config';
import { TvShowResources } from './tvShowResources';
import { Attract } from './attract';
import { Playing } from './playing';
import type { PhoneEvents } from '../phoneEvents';

export class Instructions extends State {
  processEvents(events: PhoneEvents): StateClass | null {
    if (events.hungup) return Attract;
    if (this.getStateTime() > Config.INSTRUCTIONS_TIMEOUT) return Playing;
    return null;
  }

  render(ctx: OffscreenCanvasRenderingContext2D): void {
    ctx.drawImage(TvShowResources.instructions['Forest'], 0, 0);
  }
}
